docs/architecture.md

```diff
@@ -0,0 +1,130 @@
+# Fleet Architecture
+
+## Overview
+
+At a Kubernetes level, a fleet consists of a hub cluster and zero or more member clusters.  Custom resources and associated controllers,
+residing both on the hub and member(s), as well as validating webhooks on the hub, govern the behavior of features of the fleet.  This
+document gives an architectural overview of those resources, controllers and validating webhooks.
+
+Fleets are configured such that each member *m* is granted access to resources within its corresponding *fleet-member-m* namespace on the
+hub.  Controllers on both the hub and the member are able to read, write and watch these resources to communicate between each-other.  The
+hub is not granted access to resources on any member, nor does any member have access to any other member.
+
+The following fleet use cases are implemented:
+
+1. **Join/Leave**: member clusters can join and leave a fleet.
+1. **Cluster resource placement (CRP)**: arbitrary Kubernetes resources can be replicated from a hub to one or more members, building upon
+   the Kubernetes [Work API](https://github.com/Azure/k8s-work-api).
+1. **Networking**: a single external L4 load balancer can front a service exported by one or more members, building on the Kubernetes
+   [Multi-Cluster Services API](https://github.com/kubernetes-sigs/mcs-api)
+   ([KEP-1645](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api)).  See
+   [Fleet Networking Architecture](https://github.com/Azure/fleet-networking/blob/master/docs/architecture.md) for more details.
+
+## Resources
+
+### Join/Leave
+
+The following resources are defined for the Join/Leave use case:
+
+1. [MemberCluster.fleet.azure.com](https://github.com/Azure/fleet/blob/master/apis/v1alpha1/membercluster_types.go#:~:text=type%20MemberCluster%20struct%20%7B)
+   (hub cluster, not namespaced)
+
+   Desired state of a member cluster (Join/Leave), cluster principal identity and heartbeat.  Administrative status and resource usage.
+   Typically owned by the Fleet RP.
+
+1. [InternalMemberCluster.fleet.azure.com](https://github.com/Azure/fleet/blob/master/apis/v1alpha1/internalmembercluster_types.go#:~:text=type%20InternalMemberCluster%20struct%20%7B)
+   (hub cluster, fleet-member-\* namespaces)
+
+   Internal mirror of MemberCluster resource, hosted in fleet-member-\* namespaces for communication with member clusters.  Owned by the
```

Comment:
IMC's spec is maintained by the hub controller but its status is maintained by the member controller